frontend/templates/product-list/sections/FilterableProductsSection/RefinementSingleSelect.tsx

```diff
@@ -69,21 +80,17 @@ type SingleSelectItemProps = {
    item: RefinementListRenderState['items'][0];
    attribute: string;
    productListType: ProductListType;
-   refine: RefinementListRenderState['refine'];
    onClose?: () => void;
+   onClick?: (newSelected: RefinementListItem) => void;
 };
 
 const SingleSelectItem = React.memo(function SingleSelectItem({
    item,
    attribute,
    productListType,
-   refine,
    onClose,
+   onClick,
 }: SingleSelectItemProps) {
-   const [isRefined, setIsRefined] = useDecoupledState(item.isRefined);
```

Comment:
I realize we can no longer use the decoupled state because the `SingleSelectItem`s can now change the `isRefined` value of their sibling components.

The consequence of dropping the decoupled state is that there is a delay between when the user clicks the button and when the button is highlighted. This might cause some users to click the button again before it highlights, which would effectively unrefine that selection. I think this is a bad UX, but I can't think of a way to circumvent it. Maybe when we figure out how to upgrade `react-instantsearch` we can use different hooks that will let each `SingleSelectIem` control it's own decoupled `isRefined` state again, while also preventing the flash of intermediate search results.